.github/workflows/pr-validation.yml

```
@@ -0,0 +1,208 @@
+        ignoreLabels: |
+          bot
+          ignore-semantic-pull-request
+    
+    - name: Check for breaking changes
+      run: |
+        if git log --oneline ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }} | grep -i "breaking\|BREAKING"; then
+          echo "::warning::This PR contains breaking changes. Please ensure proper versioning and documentation."
+        fi
+    
+    - name: Validate branch naming
+      run: |
```

actionlint (1.7.7)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions for more details

(expression)